### DISCUSSION TITLE
Multiple inputs/outputs per node

### DISCUSSION THREAD

**Discussion by Industrial** [xyflow/xyflow#1808] [Q&A]
Hi. I want to create an application for [Flow Based Programming](https://en.wikipedia.org/wiki/Flow-based_programming).

![Example](https://upload.wikimedia.org/wikipedia/commons/6/65/FBP_-_Simple_network.png)

This means that each Node in the graph would need to support multiple input and output ports. These ports should be named / identified. An edge should go from one node's port to another node's port.

Here is a very simple example of a program I made with my FBP runtime (Using Deno):

![42](https://github.com/Industrial/deno-fbp/blob/master/examples/42/graphviz.png?raw=true)

If this will not work with this library (which would be too bad, it looks very nice and has lots of options) is there something else I could use to implement this?

gr,

Tom

**Comment by Industrial** [comment]
Never mind. I have just found https://reactflow.dev/docs/api/node-types/

**Reply by moklick** [reply]
Great :) The custom node example also shows an implementation with multiple outputs https://reactflow.dev/examples/custom-node/
